### PR TITLE
Add some diagnostics for EFS tests

### PR DESCRIPTION
**Pull request by danmoseley**
Relates to https://github.com/dotnet/runtime/issues/12339

Dump relevant recent events and service status.
Make outerloop to minimise noise.

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                {
+                    if (description.Contains(term, StringComparison.OrdinalIgnoreCase))
+                        goto next;
+                }
+
+                _output.WriteLine($"{record.TimeCreated} {record.ProviderName} [{record.LevelDisplayName} {record.Id}] {description.Replace("\r\n", "  ")}");
+
+            next:
+                record = eventReader.ReadEvent();
+            }
```
Comment (adamsitnik) [inline src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:80]: nit: I think that we can avoid using `goto` here:

```suggestion
            var garbage = new string[] { "Background Intelligent", "Intel", "Defender", "Intune", "BITS", "NetBT"};

            _output.WriteLine("=====  Dumping recent relevant events: =====");
            
            EventRecord? record = null;
            while ((record = eventReader.ReadEvent()) != null)
            {
                string description = "";
                try
                {
                    description = record.FormatDescription();
                }
                catch (EventLogException) { }

                if (description == "" || !garbage.Any(term => description.Contains(term, StringComparison.OrdinalIgnoreCase))
                {
                    _output.WriteLine($"{record.TimeCreated} {record.ProviderName} [{record.LevelDisplayName} {record.Id}] {description.Replace("\r\n", "  ")}");
                }
            }
```

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        partial void LogEFSDiagnostics()
+        {
+            try
+            {
+                using var sc = new ServiceController("EFS");
+                _output.WriteLine($"EFS service is: {sc.Status}");
+                if (sc.Status != ServiceControllerStatus.Running)
+                {
+                    _output.WriteLine("Trying to start EFS service");
+                    sc.Start();
```
Comment (adamsitnik) [inline src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:25]: So after we run the test and it fails, we check whether the EFS service is running and if it's not, we are restarting it.

@danmoseley perhaps we should try to start running it before we run the test? or run the test only when the service is already running similar to [local file share tests](https://github.com/dotnet/runtime/blob/579ef5be313aea703818e2225b70cb7714a90cf3/src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs#L82-L99) ?

**Review by adamsitnik — COMMENTED** [review]
@danmoseley thank you for improving our test suite! 👍

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs (hunk 2 above)
Comment (danmoseley) [reply src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:26]: Applied both feedback.

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            catch (Exception e)
+            {
+                _output.WriteLine(e.ToString());
+            }
+        }
+
+        partial void LogEFSDiagnostics()
+        {
+            var hours = 1; // how many hours to look backwards
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:38]:
```suggestion
            int hours = 1; // how many hours to look backwards
```

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            var query = @$"
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:39]:
```suggestion
            string query = @$"
```

File: src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                            </Select>
+                            <Suppress Path='System'>
+                                *[System[TimeCreated[timediff(@SystemTime) &gt;= {hours * 60 * 60 * 1000L}]]]
+                            </Suppress>
+                          </Query>
+                        </QueryList> ";
+
+            var eventQuery = new EventLogQuery("System", PathType.LogName, query);
+
+            var eventReader = new EventLogReader(eventQuery);
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs:59]:
```suggestion
            using var eventReader = new EventLogReader(eventQuery);
```

**Comment by danmoseley** [comment]
any other feedback?

**Comment by danmoseley** [comment]
@stephentoub ok to merge?

**Review by stephentoub — APPROVED** [review]

**Pull request merged**